core/server/services/auth/api-key/content.js

```diff
@@ -0,0 +1,46 @@
+const models = require('../../../models');
+const {
+    BadRequestError,
+    UnauthorizedError
+} = require('../../../lib/common/errors');
+
+const authenticateContentApiKey = function authenticateContentApiKey(req, res, next) {
+    if (req.headers && req.headers.authorization) {
+        return next(new BadRequestError({
+            message: 'Content API does not support header authentication',
+            code: 'INVALID_AUTH_TYPE'
+        }));
+    }
+
+    // allow fallthrough to other auth methods or final ensureAuthenticated check
+    if (!req.query || !req.query.content_key) {
+        return next();
+    }
+
+    let key = req.query.content_key;
+
+    models.ApiKey.findOne({secret: key}).then((apiKey) => {
+        if (!apiKey) {
+            return next(new UnauthorizedError({
+                message: 'Unknown Content API Key',
+                code: 'UNKNOWN_CONTENT_API_KEY'
+            }));
+        }
+
+        // TODO: extra messaging in case Admin secret was exposed?
```

Comment:
Mm yeah - maybe we make a separate ticket for this? There is probably lots of nice lil things we could do similar to this :+1: